straitlets/builtin_models.py

```diff
@@ -58,11 +59,27 @@ def netloc(self):
         host_port = join_filter_empty(':', self.hostname, self.port)
         return '@'.join([user_pass, host_port])
 
+    query = Dict(
+        allow_none=True,
+        default_value=None,
+        help="Connection parameters",
+    )
+
+    @property
+    def query_string(self):
+        if self.query:
+            return "&".join("%s=%s" % (k, v) for k, v in iteritems(self.query))
```

Comment:
I think this is just `urlencode` from the stdlib? https://docs.python.org/2/library/urllib.html#urllib.urlencode (Note that it's been moved in Py3, so we'd have to add a compat shim for it.)